control-plane-operator/controllers/hostedcontrolplane/kas/service.go

```diff
@@ -74,6 +74,13 @@ func ReconcileService(svc *corev1.Service, strategy *hyperv1.ServicePublishingSt
 		} else {
 			svc.Spec.Type = corev1.ServiceTypeClusterIP
 		}
+		if hcp.Spec.Platform.Type == hyperv1.OpenStackPlatform && hcp.Spec.Platform.OpenStack != nil && hcp.Spec.Platform.OpenStack.KASPortID != "" {
+			// https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/expose-applications-using-loadbalancer-type-service.md#service-annotations
+			svc.Annotations["loadbalancer.openstack.org/port-id"] = hcp.Spec.Platform.OpenStack.KASPortID
+			// the loadbalancer VIP won't be associated with a floating IP.
+			// This is a hack now for the PoC; we need to design it properly, because this is very opinionated.
+			svc.Annotations["service.beta.kubernetes.io/openstack-internal-load-balancer"] = "true"
```

Comment:
we could also set `loadbalancer.openstack.org/hostname` so we get a control plane endpoint hostname instead of IP. Customers would preprovision the KAS VIP port and its DNS record before deploying the HCP.

Reply:
What would be the advantage of this?

Reply:
I can name many benefits of using a hostname vs an IP but here are a few from the top of my head:
* allow to change the underlying IP address of the Kubernetes API server without requiring reconfiguration of clients
* security policies and firewall rules are easier to manage when using DNS-based allowlists instead of IP-based rules
* (to be verified) easier to handle TLS certificates